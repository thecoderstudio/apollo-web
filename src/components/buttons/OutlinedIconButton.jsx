import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import OutlinedButton from './OutlinedButton';
import media from '../../util/media';

const propTypes = {
  onClick: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
  iconClassName: PropTypes.string.isRequired
};

const ButtonContent = styled.span`
  margin-left: 8px;

  ${media.phone`
    display: none;
  `}
`;

function OutlinedIconButton(props) {
  const {
    className, iconClassName, onClick, children
  } = props;
  return (
    <OutlinedButton className={className} onClick={onClick}>
      <i className={iconClassName} />
      <ButtonContent>{children}</ButtonContent>
    </OutlinedButton>
  );
}

OutlinedIconButton.propTypes = propTypes;

export default OutlinedIconButton;
