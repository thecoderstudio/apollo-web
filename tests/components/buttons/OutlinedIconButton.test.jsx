import React from 'react';
import renderer from 'react-test-renderer';
import OutlinedIconButton from '../../../src/components/buttons/OutlinedIconButton';

function getComponent(props) {
  return renderer.create(
    <OutlinedIconButton {...props} />
  );
}

describe('OutlinedIconButton', () => {
  const spy = jest.fn();
  let props = { children: <div/>, onClick: spy, iconClassName: "fa-plus" };

  it('renders correctly', () => {
    const tree = getComponent(props).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls onClick correctly', () => {
    const instance = getComponent({ ...props }).root;
    instance.findByType('button').props.onClick();
    expect(spy).toHaveBeenCalled();
  });
});