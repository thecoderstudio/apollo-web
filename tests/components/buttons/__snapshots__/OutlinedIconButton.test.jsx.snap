// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutlinedIconButton renders correctly 1`] = `
.c0 {
  max-height: 50px;
  padding: 12px;
  border: none;
  border-radius: 5px;
  color: white;
  font-family: 'B612',sans-serif;
  font-size: 1rem;
  font-weight: 700;
  cursor: pointer;
  background-color: transparent;
  border: 1px solid;
}

.c0:hover {
  opacity: 0.90;
}

.c0:focus {
  outline: none;
}

.c0:active {
  opacity: 0.80;
}

.c0:disabled:hover {
  opacity: 1;
  cursor: auto;
}

.c1 {
  margin-left: 8px;
}

@media (max-width:47.9375em) {
  .c1 {
    display: none;
  }
}

<button
  className="c0"
  onClick={[MockFunction]}
>
  <i
    className="fa-plus"
  />
  <span
    className="c1"
  >
    <div />
  </span>
</button>
`;
